### ISSUE TITLE
Change default heading level in Alert component from H3 to H2

### ISSUE THREAD

**Issue by davidakennedy** [CMSgov/design-system#880]
**Describe the bug**
The default heading in the Alert component is a H3. While technically, there's nothing wrong with this, I see in a lot of accessibility reviews where the heading level isn't changed as it should be to account for a better hierarchy. This results in situations where the hierarchy goes from H1 to H3. In most situations, a H2 would suffice as the default.

**To Reproduce**
1. Go to `https://design.cms.gov/example/components.alert.variations/`
2. View the heading in the Alert component

**Expected behavior**
A better default heading value that results in less heading hierarchy issues.

**Comment by bernardwang** [comment]
Sorry I'm a bit late to this, but @davidakennedy do you have examples of when an Alert should be an `h2`? Generally it's up to the consuming application to update `headingLevel`, and so I'm wary of changing the default unless it's super common that it should be `h2`. 

A change to the default would result in snapshot changes and could potentially result in heading structure issue for apps that correctly are using alert with an `h3`

**Comment by line47** [comment]
The way I was reading this was the default set to H3 often causes hierarchy issues that @davidakennedy has seen
> I see in a lot of accessibility reviews where the heading level isn't changed as it should be to account for a better hierarchy.

I'm guessing this results in the headings going from H1 to H3 with no H2 because of this default.

**Comment by bernardwang** [comment]
Right, that's my understanding as well. The problem is it's hard to gauge what's the most common heading level Alerts should be set to, so a change like this is tricky. If teams are correctly using the default `h2` heading, this change would break their hierarchy too.

**Comment by davidakennedy** [comment]
@bernardwang @line47 I see a lot of this come up where the page title is a h1, followed by an alert that's a h3. I can pull up a handful of tickets where this happens. I can definitely understand not wanting to change it. But either way, you'd likely have some sort of heading hierarchy issue. My hypothesis is that if we change this to a h2, we'll have fewer issues than we currently do.

I wonder if we could wade into some code bases to see how many h3s we have and whether they're valid.

**Comment by line47** [comment]
I like the idea of exploring how they're being used in the code currently! I think this could give us an opportunity to start thinking about what components are used and how by product teams. 🎉

**Comment by bernardwang** [comment]
Ah I see, yeah that does sound like a common usage for the Alert. I guess I'd feel more comfortable with this change if we had some data, but I'm not sure how feasible that is. It would literally mean digging through individual codebases and evaluating each Alert. 

Would it maybe be possible to ping designers on teams individually to quickly confirm Alerts are mainly used under the h1?

**Comment by davidakennedy** [comment]
We have an Alerts audit on the backlog. I wonder if we could fold this into that, assuming we're going to do it? It would hold up an answer on this though.
